### PR TITLE
.tar.xz Extraction support

### LINKED ISSUE #1766: Phoenicis failed to untar .tar.xz
### Expected behavior
Extractor extracts .tar.xz
### Actual behavior
It does not:
```
org.phoenicis.tools.archive.ArchiveException: Unable to extract the file
	at org.phoenicis.tools.archive.Tar.uncompress(Tar.java:199)
	at org.phoenicis.tools.archive.Tar.uncompressTarFile(Tar.java:89)
	at org.phoenicis.tools.archive.Extractor.uncompress(Extractor.java:69)
	at org.phoenicis.tools.archive.Extractor.uncompress(Extractor.java:45)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$1019$3124$\^eval\_$cu1$restOf.extract-1([utils, functions, filesystem, extract]:140)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$1011$395$\^eval\_$cu1$restOf.faudio([engines, wine, verbs, faudio]:22)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$1003$812AA$\^eval\_.run#L:23(<eval>:24)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$933$553$\^eval\_$cu1$restOf.go([engines, wine, quick_script, installer_script]:75)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$868$259$\^eval\_.run(<eval>:11)
	at jdk.nashorn.javaadapters.org_phoenicis_scripts_Installer.run(Unknown Source)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$867$\^eval\_$cu1$restOf.:program(<eval>:33)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:655)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:513)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:527)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:456)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:413)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:409)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:162)
	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
	at org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
	at org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.IOException: Error detected parsing the header
	at org.apache.commons.compress.archivers.tar.TarArchiveInputStream.getNextTarEntry(TarArchiveInputStream.java:285)
	at org.apache.commons.compress.archivers.tar.TarArchiveInputStream.getNextEntry(TarArchiveInputStream.java:552)
	at org.phoenicis.tools.archive.Tar.uncompress(Tar.java:160)
	... 24 more
Caused by: java.lang.IllegalArgumentException: Invalid byte 94 at offset 0 in '^ vP��J?' len=8
	at org.apache.commons.compress.archivers.tar.TarUtils.parseOctal(TarUtils.java:141)
	at org.apache.commons.compress.archivers.tar.TarUtils.parseOctalOrBinary(TarUtils.java:171)
	at org.apache.commons.compress.archivers.tar.TarArchiveEntry.parseTarHeader(TarArchiveEntry.java:1277)
	at org.apache.commons.compress.archivers.tar.TarArchiveEntry.parseTarHeader(TarArchiveEntry.java:1266)
	at org.apache.commons.compress.archivers.tar.TarArchiveEntry.<init>(TarArchiveEntry.java:404)
	at org.apache.commons.compress.archivers.tar.TarArchiveInputStream.getNextTarEntry(TarArchiveInputStream.java:283)
	... 26 more
```
### Steps to reproduce
First steps of the FAudio verb [https://github.com/PhoenicisOrg/scripts/pull/804](https://github.com/PhoenicisOrg/scripts/pull/804)
### System information
* software version (git revision) of phoenicis and scripts repositories: 7347f941e004934f42c622013406a2e4ddb994ec
* used operating system Lubuntu 18.10
* used Java version 
```
openjdk 11.0.1 2018-10-16
OpenJDK Runtime Environment (build 11.0.1+13-Ubuntu-2ubuntu1)
OpenJDK 64-Bit Server VM (build 11.0.1+13-Ubuntu-2ubuntu1, mixed mode, sharing)
```

### PR DESCRIPTION
**Pull request by qparis**
Fixes #1766 

Note: I have duplicated magic xml files to modify them until they validate my merge request: https://github.com/arimus/jmimemagic/pull/39

### CODE REVIEW

**Comment by qparis** [comment]
@ImperatorS79 Should fix your problem

**Comment by ImperatorS79** [comment]
Working for me.

File: phoenicis-tools/pom.xml [hunk 1]
```diff
@@ -59,9 +60,9 @@
             <version>2.6</version>
         </dependency>
         <dependency>
-            <groupId>jmimemagic</groupId>
+            <groupId>net.sf.jmimemagic</groupId>
```
Comment (madoar) [inline phoenicis-tools/pom.xml]: Is this a snapshot repository?

**Comment by plata** [comment]
@qparis are you sure about Codacy? I think if it is not fixed now, it might popup all the time later.

File: phoenicis-tools/src/main/java/org/phoenicis/tools/http/Downloader.java [hunk 2]
```diff
@@ -256,6 +256,7 @@ private void get(URL url,
                 connection.setHeaders(headers);
             }
 
+            connection.connect();
```
Comment (plata) [inline phoenicis-tools/src/main/java/org/phoenicis/tools/http/Downloader.java]: Looks like you should merge master into the branch.

**Comment by qparis** [comment]
> @qparis are you sure about Codacy? I think if it is not fixed now, it might popup all the time later.

Yes, the test is delegated in another method, so no worries

File: phoenicis-tools/pom.xml (hunk 1 above)
Comment (qparis) [reply phoenicis-tools/pom.xml]: Don't think so

File: phoenicis-tools/src/main/java/org/phoenicis/tools/http/Downloader.java (hunk 2 above)
Comment (qparis) [reply phoenicis-tools/src/main/java/org/phoenicis/tools/http/Downloader.java]: Solved

**Comment by plata** [comment]
Shall we wait until your changes are available in jmimemagic or do you want to use the code here? If so, we must remember to remove it later. Is it ok to do so from a license perspective?

**Comment by qparis** [comment]
https://github.com/arimus/jmimemagic/blob/master/LICENSE -> Apparently yes 

I have made another modification that prevents jmimemagic from being mixed up with .zip files. If my merge request is accepted, I will propose it also. 

In the mean time, let's merge this request because I'm not totally sure that the project is very active

**Comment by madoar** [comment]
@qparis does it make sense to fork the repository, make the changes on our fork and then include it via https://jitpack.io/?

**Comment by qparis** [comment]
Not totally sure yet... 
I think there is a very huge room of improvement to enhance this library so I don't know if we should fork it completely or not.

Also, I don't know what is its memory footprint

Any thought?

**Comment by qparis** [comment]
I agree. If my merge request is not merge, we fork the project. Do you agree with that?

**Comment by madoar** [comment]
Yes ;)

**Comment by plata** [comment]
There's no active Maven dependency with this functionality?

**Pull request merged**